pkg/api/mlflow/service/artifact/storage/local_test.go

```diff
@@ -7,8 +7,83 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/config"
 )
 
+// TestGetArtifact for the local storage implementation.
+func TestGetArtifact_Ok(t *testing.T) {
+	// setup
+	defaultArtifactRoot := t.TempDir()
+	runArtifactRoot := "/run-artifact-root/"
+	fileName := "file.txt"
+	fileContent := "artifact content"
+
+	err := os.MkdirAll(filepath.Join(defaultArtifactRoot, runArtifactRoot), os.ModePerm)
+	assert.Nil(t, err)
+
+	f, err := os.Create(filepath.Join(defaultArtifactRoot, runArtifactRoot, fileName))
+	assert.Nil(t, err)
+	_, err = f.Write([]byte(fileContent))
+	assert.Nil(t, err)
+
+	// invoke
+	svcConfig := &config.ServiceConfig{
+		DefaultArtifactRoot: defaultArtifactRoot,
+	}
+	svc, err := NewLocal(svcConfig)
+	assert.Nil(t, err)
+
+	file, err := svc.GetArtifact(filepath.Join(defaultArtifactRoot, runArtifactRoot), fileName)
+	assert.Nil(t, err)
+	defer func() {
+		file.Close()
+		os.Remove(f.Name())
+	}()
+
+	// verify
+	assert.NotNil(t, file)
+	p := make([]byte, 20)
+	ln, err := file.Read(p)
+	assert.Nil(t, err)
+
+	assert.Equal(t, fileContent, string(p[:ln]))
```

Comment:
if you just check `fileContent` with what `p` has won't it be the same?

Reply:
p will potentially have extra bytes